processing/src/main/java/io/druid/query/MetricsEmittingMergingBlockingPool.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query;
+
+import io.druid.collections.DefaultBlockingPool;
+import io.druid.java.util.emitter.service.ServiceEmitter;
+import io.druid.java.util.emitter.service.ServiceMetricEvent;
+import com.google.common.base.Supplier;
+
+public class MetricsEmittingMergingBlockingPool<T> extends DefaultBlockingPool<T>
+    implements ProcessingMonitor.MetricEmitter
+{
+  public MetricsEmittingMergingBlockingPool(
+          Supplier<T> generator,
+          int limit,
+          ProcessingMonitor processingMonitor)
```

Comment:
Code style is a bit off here (the ending paren should be on its own line; also, indentation). If you are using IntelliJ, you can apply Druid's code style from the codestyle config in the repo. Also, our checkstyle should catch some things like this.